addons/point_of_sale/static/src/js/Popups/ClosePosPopup.js

```diff
@@ -0,0 +1,172 @@
+                this.moneyDetailsRef.comp.openPopup();
+                this.state[this.defaultCashDetails.name].counted = 0;
+                if (this.manualInputCashCount) {
+                    this.moneyDetailsRef.comp.reset();
+                }
+            }
+        }
+        handleInputChange(paymentName) {
+            let expectedAmount;
+            if (paymentName === this.defaultCashDetails.name) {
+                this.manualInputCashCount = true;
+                expectedAmount = this.defaultCashDetails.amount;
+                this.state.notes = '';
+            } else {
+                expectedAmount = this.otherPaymentMethods.find(paymentMethod => paymentMethod.name == paymentName).amount;
+            }
+            this.state[paymentName].difference = this.state[paymentName].counted - expectedAmount;
+        }
+        updateCountedCash(event) {
+            const { total, moneyDetailsNotes, moneyDetails } = event.detail;
+            const rounding = this.env.pos.currency.rounding;
+            this.state[this.defaultCashDetails.name].counted = round_pr(total, rounding);
+            this.state[this.defaultCashDetails.name].difference =
+                round_pr(this.state[[this.defaultCashDetails.name]].counted - this.defaultCashDetails.amount, rounding);
+            if (moneyDetailsNotes) {
+                this.state.notes = moneyDetailsNotes;
+            }
+            this.manualInputCashCount = false;
+            this.moneyDetails = moneyDetails;
+        }
+        canCloseSession() {
+            return !this.cashControl || !this.state[this.defaultCashDetails.name].difference || this.state.acceptClosing;
+        }
+        closePos() {
+            this.trigger('close-pos');
+        }
+        async closeSession() {
+            if (this.canCloseSession() && !this.closeSessionClicked) {
+                this.closeSessionClicked = true;
+                let response;
+                if (this.cashControl) {
+                     response = await this.rpc({
+                        model: 'pos.session',
+                        method: 'post_closing_cash_details',
+                        args: [this.env.pos.pos_session.id],
+                        kwargs: {
+                            counted_cash: this.manualInputCashCount ? this.state[this.defaultCashDetails.name].counted : null,
+                            bill_details: this.manualInputCashCount ? null : Object.entries(this.moneyDetails).filter(money => money[1])
+                        }
+                    })
+                    if (!response.successful) {
+                        return this.handleClosingError(response);
+                    }
+                }
+                await this.rpc({
+                    model: 'pos.session',
+                    method: 'update_closing_control_state_session',
+                    args: [this.env.pos.pos_session.id, this.state.notes]
+                })
```

Comment:
Should this also be inside the try-catch statement below? From what I understand, the catch clause is able to handle the error resulting from this rpc call. Or is there something I'm missing?

Reply:
Nah because if there's an error, I don't want the user to be redirected to the back end (even for the first RPC call)